### PR TITLE
[DimExpr] Implement DimExprBuilder interface

### PR DESCRIPTION
**Pull request by jiahy0825**
### PR types
New features

### PR changes
Others

### Description

pcard-76996

[DimExpr] Implement DimExprBuilder interface

### CODE REVIEW

File: paddle/pir/dialect/shape/utils/dim_expr_builder.cc [hunk 1]
```diff
@@ -56,7 +57,12 @@ DimExpr DimExprBuilder::Broadcast(const DimExpr& lhs, const DimExpr& rhs) {
 
 std::vector<DimExpr> DimExprBuilder::ConstShape(
     const std::vector<std::int64_t>& dims) {
```
Comment (jiahy0825) [inline paddle/pir/dialect/shape/utils/dim_expr_builder.cc:59]: 提供了从 std::vector<std::int64_t> 构造 vector<DimExpr> 的接口

File: paddle/pir/dialect/shape/utils/dim_expr_builder.cc [hunk 2]
```diff
@@ -74,21 +80,38 @@ void DimExprBuilder::CstrEq(const DimExpr& lhs, const DimExpr& rhs) {
...
+  EmplaceDimExpr(lhs);
+  EmplaceDimExpr(rhs);
+  return ret;
 }
 
 std::pair<std::vector<DimExpr>, std::vector<DimExpr>> DimExprBuilder::SplitAt(
-    const std::vector<DimExpr>, int index) {
-  SYMBOL_NOT_IMPLEMENTED;
+    const std::vector<DimExpr> dim_exprs, int index) {
+  std::vector<DimExpr> lhs(dim_exprs.begin(), dim_exprs.begin() + index);
```
Comment (tc20042008) [inline paddle/pir/dialect/shape/utils/dim_expr_builder.cc:113]: 检查index合法性

**Review by lanxianghit — DISMISSED** [review]

**Review by zhangboSJTU — DISMISSED** [review]
LGTM

File: paddle/pir/dialect/shape/utils/dim_expr_builder.cc (hunk 2 above)
Comment (jiahy0825) [reply paddle/pir/dialect/shape/utils/dim_expr_builder.cc:113]: done, thanks

**Review by tc20042008 — APPROVED** [review]

**Pull request merged**